app/services/subscriber_list_params/generate_single_page_list_params_service.rb

```diff
@@ -0,0 +1,47 @@
+module SubscriberListParams
+  class GenerateSinglePageListParamsService < ApplicationService
+    DOCUMENT_TYPES_WITH_REVERSE_LINKED_LISTS = { "document_collection" => "document_collections" }.freeze
+
+    def initialize(content_item)
+      super()
+      @content_item = content_item
+    end
+
+    def call
+      {
+        "url" => content_item["base_path"],
+        "title" => content_item["title"],
+        "content_id" => content_item["content_id"],
+        "description" => content_item["description"],
+      }.merge(strip_if_empty_value(links))
```

Comment:
I'm unsure of the need for the `strip_if_empty_value` and `strip_if_empty_key` methods.

I think these checks can probably be done in the `links` method. 

The check here can be replaced with an `present?` check, if the hash is assigned to a variable first.

Reply:
Thanks Leena - not sure why I made that so complicated. Have refactored 👍